--- src/main/java/com/xpn/xwiki/render/macro/ImageMacro.java	2006-07-12 14:09:57.000000000 +0300
+++ src/main/java/com/xpn/xwiki/render/macro/ImageMacro.java	2006-07-27 12:13:51.000000000 +0300
@@ -56,10 +56,10 @@
     str.append("<img src=\"");
     str.append(doc.getAttachmentURL(img, "download", xcontext));
     str.append("\" ");
-    if ((!"none".equals(height))&&(height!=null))
-     str.append("height=\"" + height + "\" ");
-    if ((!"none".equals(width))&&(width!=null))
-           str.append("width=\"" + width + "\" ");
+    if ((!"none".equals(height))&&(height!=null)&&(!"".equals(height.trim())))
+     str.append("height=\"" + height.trim() + "\" ");
+    if ((!"none".equals(width))&&(width!=null)&&(!"".equals(width.trim())))
+           str.append("width=\"" + width.trim() + "\" ");
     str.append("alt=\"");
     str.append(img);
     str.append("\" />");
