Index: Cache.java
===================================================================
RCS file: /cvs/oscache/src/java/com/opensymphony/oscache/base/Cache.java,v
retrieving revision 1.5
diff -u -r1.5 Cache.java
--- Cache.java	9 Apr 2006 12:50:34 -0000	1.5
+++ Cache.java	30 Jul 2006 20:19:07 -0000
@@ -83,6 +83,12 @@
     private boolean blocking = false;
 
     /**
+     *  Optional cache name which allows to be used to identify a cache
+     *  in SharedJavaGroupsBroadcastingListener and direct the notifications to the right cache
+     */
+    private String name = "";
+
+    /**
      * Create a new Cache
      *
      * @param useMemoryCaching Specify if the memory caching is going to be used
@@ -142,7 +148,22 @@
 
         this.blocking = blocking;
     }
-    
+
+    /*
+     * @return the name of the cache
+     */
+    public String getName() {
+        return name;
+    }
+
+    /*
+     * Sets the name of the cache
+     * @name name to be set
+     */
+    public void setName(String name) {
+        this.name = name;
+   }
+
     /**
      * @return the maximum number of items to cache can hold.
      */
